src/commons/listener.rs

```diff
@@ -7,7 +7,9 @@ use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
 
 #[cfg(doc)]
-use k8s_openapi::api::core::v1::{Node, Pod, Service, Volume};
+use k8s_openapi::api::core::v1::{
+    Node, PersistentVolume, PersistentVolumeClaim, Pod, Service, Volume,
+};
```

Comment:
It'd make me really happy if we had a module-level docstring as well for this file (although slightly out of scope for this PR). A general destriction of what the structs are for, which other components interact with them, and how they are related would be great. maybe also a sentence or two about the underlying motivation.

Reply:
I can try to add a bit, but most of these are also things that should probably go in the user-facing Antora docs...

Reply:
6d33001232177474309398984ea811661ae0fa8f

Reply:
perfect, thanks :ok_hand: